clients/python-wrapper/tests/utests/test_branch.py

```diff
@@ -157,9 +157,9 @@ def monkey_get_commit(repo_name, ref_name, **_):
             branch.revert(None)
 
         # both are passed, prefer ``reference_id``
-        with pytest.warns(DeprecationWarning, match="reference_id is deprecated.*"):
+        with pytest.raises(ValueError, match="`reference_id` and `reference` cannot both be provided.*"), \
+                pytest.warns(DeprecationWarning, match="reference_id is deprecated.*"):
             # this is not a high-quality test, but it would throw if the revert API
             # was called with reference "hello" due to the monkey-patching above
             # always returning "ab1234" as ref ID.
-            c = branch.revert(ref_id, reference_id="hello")
-            assert c.id == ref_id
+            branch.revert(ref_id, reference_id="hello")
```

Comment:
This mixes up two cases.  Please use a separate test.  For instance, if we _always_ raise that ValueError when `reference_id` is provided, your modified test would still pass!

Reply:
This test asserts that when both args are provided, an error message and a warning will be raised - this is one case.
If we decide to always raise an error when `reference_id` is provided, [this test](https://github.com/treeverse/lakeFS/blob/master/clients/python-wrapper/tests/utests/test_branch.py#L152) will fail.